File: src/JavaScript/Chat/Generate.js
import { execFile } from 'child_process';
import { mkdirSync, existsSync } from 'fs';
import path from 'path';

// Define paths
const antlrJar = path.resolve('src/JavaScript/Chat/libs/antlr-4.13.2-complete.jar');
const grammarFile = path.resolve('src/JavaScript/Chat/Chat.g4');
const outputDir = path.resolve('CompiledFile');

// Ensure output directory exists
if (!existsSync(outputDir)) {
    mkdirSync(outputDir, { recursive: true });
}

// Build the ANTLR command
const command = 'java';
const args = [
    '-jar', antlrJar,
    '-Dlanguage=JavaScript',
    '-o', outputDir,
    grammarFile
];

// Run the command
execFile(command, args, (error, stdout, stderr) => {
    if (error) {
        console.error(`Error running ANTLR: ${error.message}`);
        return;
    }
    if (stderr) {
        console.error(`ANTLR stderr:\n${stderr}`);
    }
    console.log(`ANTLR generation successful. Files saved to '${outputDir}'`);
});

File: src/Python/Chat/Generate.py
import subprocess
import os

# Define paths
antlr_jar = os.path.abspath("libs/antlr-4.13.2-complete.jar")
grammar_file = "Chat.g4"
output_dir = "CompiledFile"

# Make sure the output directory exists
os.makedirs(output_dir, exist_ok=True)

# Build the ANTLR command
command = [
    "java", "-jar", antlr_jar,
    "-Dlanguage=Python3",
    "-o", output_dir,
    grammar_file
]

# Run the command
try:
    subprocess.run(command, check=True)
    print(f"ANTLR generation successful. Files saved to '{output_dir}'")
except subprocess.CalledProcessError as e:
    print(f"Error running ANTLR: {e}")